apps/consumer-client/pages/examples/prove-and-add.tsx

```diff
@@ -0,0 +1,101 @@
+import { constructPassportPcdProveAndAddRequestUrl } from "@pcd/passport-interface";
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import { SemaphoreGroupPCDPackage } from "@pcd/semaphore-group-pcd";
+import { SemaphoreSignaturePCDPackage } from "@pcd/semaphore-signature-pcd";
+import { HomeLink } from "../../components/Core";
+import { ExampleContainer } from "../../components/ExamplePage";
+import { PASSPORT_URL, SEMAPHORE_GROUP_URL } from "../../src/constants";
+import { sendPassportRequest } from "../../src/util";
+
+export default function Page() {
+  return (
+    <div>
+      <HomeLink />
+      <h2>Prove and Add</h2>
+      <ExampleContainer>
+        <button onClick={addGroupMembershipProof}>
+          prove and add a group membership proof
+        </button>
+        <br />
+        <br />
+        <button onClick={addSignatureProof}>
+          prove and add a signature proof
+        </button>
+      </ExampleContainer>
+    </div>
+  );
+}
+
+async function addGroupMembershipProof() {
+  const url = constructPassportPcdProveAndAddRequestUrl<
+    typeof SemaphoreGroupPCDPackage
+  >(
+    PASSPORT_URL,
+    window.location.origin + "/popup",
+    SemaphoreGroupPCDPackage.name,
+    {
```

Comment:
can this JSON object live in semaphoregrouppcd? (or at least the field names and types. then, the caller could specify a partial object if they want some of them to be not user-provided.) feels like a lot to copy around

Reply:
I personally kinda prefer it to be explicitly defined every time so the dev does have to think about it.